rust/forevervm/src/commands/auth.rs

```diff
@@ -21,6 +26,68 @@ pub async fn whoami() -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn signup(base_url: Url) -> anyhow::Result<()> {
+    let config_manager = ConfigManager::new()?;
+    let config = config_manager.load()?;
+    if config.token.is_some() {
+        println!("Already logged in");
+        return Ok(());
+    }
+
+    println!(
+        "Enter your email and an account name below, and we'll send you a ForeverVM API token!\n"
+    );
+
+    let email = Input::new()
+        .with_prompt("Enter your email")
+        .allow_empty(false)
+        .validate_with(|input: &String| -> Result<(), &str> {
+            if util::validate_email(input) {
+                Ok(())
+            } else {
+                Err("Are you sure that's a valid email address?")
```

Comment:
nit but i'd avoid this sort of jokey tone in error messages. feels a bit like rubbing salt in a paper cut

Reply:
Haha, I hear ya. I started with `Invalid email` and then figured that was a little too sterile. How about this?:

> Please enter a valid email address (example: name@company.com)